<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <a class="navbar-brand" href="0_Dataoving1_Forside.ipynb">Dataøving 1</a>
    </div>
    <ul class="nav navbar-nav">
        <li><a href="Oppgave1.ipynb">Oppgave 1 - Komplekse Signal</a></li>
        <li><a href="Oppgave2.ipynb">Oppgave 2 - Generering og analyse av C-dur skala</a></li>
        <li class="active"><a href="Oppgave3.ipynb">Oppgave 3 - Amplitudemodulasjon</a></li>
        <li><a href="Oppgave4.ipynb">Oppgave 4 -Frekvensdeteksjon</a></li>
    </ul>
  </div>
</nav>

# Amplitudemodulasjon

__Læringsmål__:
* Forståelse for konseptet frekvensforskyvning
* Effekten av frekvensmiksing på frekvensinnholdet til et signal

__Bibliotek og notebook-konfigurasjon:__

In [1]:
from numpy import sin, cos, pi, exp, empty  # Sentrale funksjoner for oppgaven
import numpy as np                          # Importer funksjonalitet fra numpy biblioteket med prefiks "np"
import matplotlib.pyplot as plt             # Importer pyplot modulen i matplotlib med prefiks "plt"


from funksjoner_data2 import displayDualSpectrum # Vedlegg til dataøving 2
%matplotlib ipympl

Amplitudemodulasjon (AM) er en gammel teknikk som ble brukt til å overføre analoge signal ved hjelp av radiobølger. Grunnkonseptet går ut på å multiplisere (mikse) det analoge signalet med en bærebølge med mye høyere frekvens, slik at man ender opp med et informasjonsbærende signal hvor spektralinnholdet er sentrert rundt carrier frekvensen . Bildet nedenfor viser et utklipp et signal, en bærebølge, og det resulterende AM signalet.

![](Figurer/Amplitudemodulasjon.png)

Resultatet blir en såkalt "Beat Note" hvor vi får én gjenkjennbar hovedfrekvens som varierer i amplitude over tid.

Vi tar utgangspunkt i et basis-signal $x(t)$ som kan beskrives matematisk som følger:
$$x(t) = A_0 + A_1 \cdot \cos(2\pi \cdot f_1 \cdot t) + A_2 \cdot \cos(2\pi \cdot f_2 \cdot t)$$
Vi skal deretter lage et amplitudemodulert signal $y(t) = x(t) \cdot cos(2\pi \cdot f_c \cdot t)$ der $f_c$ er carrier-frekvensen. Parametrene settes i kodecellen nedenfor.

In [2]:
Fs = 240 # Samplingsfrekvens
F1 = 8
F2 = 12
A0 = 0.48
A1 = 0.31
A2 = 0.19
Fc = 50 #Carrier Frekvens
T = 20  #Signalets varighet i sekund

## a)
* Fullfør koden i cellen nedenfor slik at vi har to signal i form av array objekter `xt` og `yt` som inneholder signalene $x(t)$ og $y(t)$ som beskrevet ovenfor. 

In [3]:
# Signalgenerasjon:

t = np.linspace(0, T, T*Fs, endpoint=False) # 

### BEGIN SOLUTION
xt = A0 + A1*cos(2*pi*F1*t)+ A2*cos(2*pi*F2*t)# Implementer riktig formel her
yt = xt*cos(2*pi*Fc*t) # Implementer riktig formel her
### END SOLUTION

# Plotting av signalspektrum:
plt.close(1)
plt.figure(1, figsize=(8,6))
plt.subplot(5,1,(1,2))
displayDualSpectrum(xt,Fs)
plt.title('Basisbånd-signal x(t)')
plt.subplot(5,1,(4,5))
displayDualSpectrum(yt,Fs)
plt.title('Frekvensmikset signal y(t)')
plt.show()

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

## b)
Prøv deg fram med å justere de ulike signalparametrene, og svar på følgende kontrollspørsmål:
* Hvorfor har vi ulik amplitude for signalkomponentene i delplot 1 og delplot 2?
* Hva avgjør _båndbredden_ til signalet $y(t)$

* Signalet har blitt frekvensforskjøvet med to komplekse eksponentialer som har amplitude lik 0.5, og motsatt rotasjonsretning. Vi får da to "kopier" av frekvensinnholdet til $x(t)$ forskjøvet i frekvens, og med halvparten av amplituden hver.
* Båndbredden til signalet $y(t)$ er avgjort av hvor bredden til det _tosidige_ frekvensspekteret til $x(t)$. Det vil si at $B = 2\cdot \max(f_1, f_2) = 24 Hz$

## c)

* Lag et nytt signal $v(t) = y(t)\cdot \cos(2\pi \cdot f_c \cdot t)$ og vis det tosidige spekteret til $v(t)$ i en ny figur. Kommenter på hva som skjer med frekvensinnholdet til det nye signalet.

In [4]:
### BEGIN SOLUTION
plt.figure(2)
vt = yt*cos(2*pi*Fc*t)
displayDualSpectrum(vt, Fs)
### END SOLUTION

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

Vi ser tydelig at frekvensinnholdet i signalet $y(t)$ splittes på nytt, og forskyves igjen i begge retnigner. Det interessante nå er at vi får en overlapping i basisbåndet igjen, noe som gir en delvis rekonstruksjon av frekvensinnholdet til $x(t)$.

## d) 
* Bruk så funksjonen `numpy.mean()` til å regne ut gjennomsnittsverdien til de genererte signalene $x(t)$, $y(t)$ og $v(t)$. Kommenter på sammenhengen med frekvensplotene fra de tidligere deloppgavene.

In [5]:
### BEGIN SOLUTION
print("x(t) average:", round(np.mean(xt), 3))
print("y(t) average:", round(np.mean(yt),3))
print("v(t) average:", round(np.mean(vt),3))
### END SOLUTION

x(t) average: 0.48
y(t) average: 0.0
v(t) average: 0.24


Konstantleddet i et signal er det samme som en frekvenskomponent der $f=0$. Dette betyr at "stolpen" i frekvensspekteret ved $f=0$ vil tilsvare gjennomsnittsverdien til signalet.

Dette stemmer veldig bra, da $x(t)$ og $v(t)$ har gjennomsnittsverdier på henholdsvis 0.48 og 0.24, noe som stemmer bra med figurene. $y(t)$ har gjennnomsnittsverdi lik 0, noe som også stemmer da frekvensinnholdet ikke viser noen frekvenskomponent ved $f=0$.